mongodb_depot.go

```diff
@@ -124,6 +124,35 @@ func (m *mongoDepot) Check(tag *depot.Tag) bool {
 	}
 }
 
+// CheckWithError returns whether the user and data specified by the tag exists
+// as well as an error in the case of an internal error.
+func (m *mongoDepot) CheckWithError(tag *depot.Tag) (bool, error) {
+	name, key, err := getNameAndKey(tag)
+	if err != nil {
+		return false, nil
+	}
+
+	u := &User{}
+
+	err = m.client.Database(m.databaseName).Collection(m.collectionName).FindOne(m.ctx, bson.D{{Key: userIDKey, Value: name}}).Decode(u)
+	if errNotNoDocuments(err) {
+		return false, errors.Wrap(err, "checking depot tag")
+	}
+
+	switch key {
+	case userCertKey:
+		return u.Cert != "", nil
+	case userPrivateKeyKey:
+		return u.PrivateKey != "", nil
+	case userCertReqKey:
+		return u.CertReq != "", nil
+	case userCertRevocListKey:
+		return u.CertRevocList != "", nil
+	default:
+		return false, nil
```

Comment:
Same question as above about invalid keys.

Reply:
This isn't an error cause it means none of the keys exist, which is fine.